model/src/main/kotlin/Hash.kt

```diff
@@ -61,7 +61,7 @@ data class Hash(
                     algorithm = HashAlgorithm.fromString(splitValue.first())
                 )
             } else {
-                Hash(value.lowercase(), HashAlgorithm.create(value))
+                Hash(value, HashAlgorithm.create(value))
```

Comment:
Maybe this should be considered a breaking change because now `Hash("ABC...")` is not equal to `Hash("abc...")` anymore, and I wonder if it should be.

Reply:
> I wonder if it should be.

I agree it should be, and we have `fun verify(hash: Hash)` for that. I guess the reason this was that we were too lazy to override `equals()` and `hashCode()`... maybe we should change that.